modules/core/src/main/java/org/apache/ignite/internal/processors/cache/QueryCursorImpl.java

```diff
@@ -92,6 +93,11 @@ public QueryCursorImpl(Iterable<T> iterExec, GridQueryCancel cancel, boolean isQ
         return new AutoClosableCursorIterator<>(this, iter());
     }
 
+    /** {@inheritDoc} */
+    @Override public Spliterator<T> spliterator() {
+        return iterExec == null ? QueryCursorEx.super.spliterator() : iterExec.spliterator();
```

Comment:
Do we have a scenario when `iterExec` is null?

Reply:
You don't use @Nonnull annotations or some `com.google.common.base.Preconditions` equivalent in this project, right? Force of habit, I guess. :) will fix.